### PR TITLE
Resolve #192 "Custom power generator (solar panel?) building guide"

### LINKED ISSUE #192: Custom power generator (solar panel?) building guide

### PR DESCRIPTION
**Pull request by ARtemachka**
Feel free to check and leave comments/suggestions.

### CODE REVIEW

**Review by budak7273 — CHANGES_REQUESTED** [review]
See the [discord](https://discord.com/channels/555424930502541343/1036634533077979146/1159671730609078312) discussion

File: modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+At this point you should be able to construct your buildable in game just like the Plantpot of Knowledge,
+and just like the plantpot, it doesn't do anything ... just yet!
+
+[TIP]
+=====
+There are also subclasses of `FGBuildableGenerator` such as
+`FGBuildableGeneratorFuel`, `FGBuildableGeneratorGeoThermal`, and `FGBuildableGeneratorNuclear`
+which, as you can judge from their names, contain some specific logic for different types of power generators.
+Since as our Solar Panel does not require the custom behaviors of any of these, we will not use them.
```
Comment (ARtemachka) [inline modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc:28]: > **_Since as_** our Solar Panel

It's redundant to have both, remove one of these.

File: modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+=====
+
+== Adding Power Generation Logic
+
+The first thing we want to add to our `Build_DocSolarPanel` is an `FGPowerConnection` component,
+as we need to connect our Solar Panel to the power lines to use its generated power to.
```
Comment (ARtemachka) [inline modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc:34]: > to use its generated power **_to_**.

Remove.

File: modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc [hunk 3]
```diff
@@ -32,58 +32,69 @@ Since as our Solar Panel does not require the custom behaviors of any of these,
...
+That might be okay if we were making an https://en.wikipedia.org/wiki/Radioisotope_thermoelectric_generator[RTG],
+but we're making a solar panel, so we want it to only work during the day.
 
-The buildable class we are extending already has code
-to associate our Power Connection with the building's internal Power Info.
-If that wasn't the case for some reason, we could use
-the `Get Power Info` node and the power connection's `Set Power Info` node to establish the connection.
+The first step to fixing this is to disable the generator's dynamic power production ability.
+Use a `Get Power Info` node and call `Set Dynamic Production Capacity` to `0.0`.
+This value defaults to the value we set in in our building,
```
Comment (ARtemachka) [inline modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc:54]: > the value we set in **_in_** our building

Remove.

File: modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc [hunk 4]
```diff
@@ -32,58 +32,69 @@ Since as our Solar Panel does not require the custom behaviors of any of these,
...
+while still correctly reporting the power production in the build gun.
+Hook up the execution pin to Begin Play, since we only need this to happen once - on actor spawn.
 
-// TODO must set the "M Power Production" property of the buildable for the buildgun to report the power
+image:BeginnersGuide/simpleMod/machines/SolarPanel/DisableDynamicProduction.png[Setting the Dynamic Production]
 
-// TODO revise this paragraph with the new way of setting power output
-Then, we need to set the amount of power our Solar Panel would produce and we can do that with PowerInfo's `Set Base Production` node, specifying the amount of, for example, 20.0.
-For now, the code should look like this:
+Next, we first need to figure out whether it's day or night and adjust power production accordingly.
```
Comment (ARtemachka) [inline modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc:61]: > Next, we **_first_** need

Remove.

File: modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc [hunk 5]
```diff
@@ -32,58 +32,69 @@ Since as our Solar Panel does not require the custom behaviors of any of these,
...
-Combining this with a `Select Float` node allows setting the base production set depending on the time of day:
+image:BeginnersGuide/simpleMod/machines/SolarPanel/EventBound.png[Binding the Event]
 
-// TODO retake new screenshot from examplemod
-image:BeginnersGuide/simpleMod/SolarPanel_2.png[image]
+Next, we need to set the power production depending on the passed time value.
+We'll be using the `Set Base Production` method of the power info to do this,
+since unlike the dynamic production, this power production is always "full blast" and will charge batteries.
+Use a Select Float node to select between a daytime production value and a night time production value of zero.
+We can avoid repeating outselves here and use the parent class `Get Default Power Production Capacity` method
```
Comment (ARtemachka) [inline modules/ROOT/pages/Development/BeginnersGuide/SimpleMod/machines/SolarPanel.adoc:79]: > avoid repeating **_outselves_** here

Ou**r**selves.

**Review by ARtemachka — COMMENTED** [review]
Everything else looks good so you can submit after these are fixed.

**Review by budak7273 — APPROVED** [review]

**Pull request merged**